ocaml-lsp-server/test/e2e/__tests__/ocamllsp-hoverExtended.ts

```diff
@@ -0,0 +1,613 @@
+      {
+        textDocument: Types.TextDocumentIdentifier.create("file:///test.ml"),
+        position: Types.Position.create(24, 4),
+        verbosity: 0,
+      },
+    );
+
+    // now the same hover as before comes with unrelated documentation
+    expect(buggedHoverOverK).toMatchInlineSnapshot(`
+      Object {
+        "contents": Object {
+          "kind": "markdown",
+          "value": "\`\`\`ocaml
+      unit
+      \`\`\`",
+        },
+        "range": Object {
+          "end": Object {
+            "character": 5,
+            "line": 24,
+          },
+          "start": Object {
+            "character": 4,
+            "line": 24,
+          },
+        },
+      }
+    `);
+  });
+
+  it("", async () => {
+    languageServer = await LanguageServer.startAndInitialize();
+    await languageServer.sendNotification("textDocument/didOpen", {
+      textDocument: Types.TextDocumentItem.create(
+        "file:///test.ml",
+        "ocaml",
+        0,
+        outdent`
+type foo = int
+
+let x : foo = 1
+        `
+      ),
+    });
+
+    let result0 = await languageServer.sendRequest("textDocument/hover", {
+      textDocument: Types.TextDocumentIdentifier.create("file:///test.ml"),
+      position: Types.Position.create(2, 4),
+      verbosity: 0,
+    });
+
+    expect(result0).toMatchInlineSnapshot(`
+      Object {
+        "contents": Object {
+          "kind": "plaintext",
+          "value": "foo",
+        },
+        "range": Object {
+          "end": Object {
+            "character": 5,
+            "line": 2,
+          },
+          "start": Object {
+            "character": 4,
+            "line": 2,
+          },
+        },
+      }
+    `);
+
+    let result1 = await languageServer.sendRequest("textDocument/hover", {
+      textDocument: Types.TextDocumentIdentifier.create("file:///test.ml"),
+      position: Types.Position.create(2, 4),
+      verbosity: 1,
+    });
+
+    expect(result1).toMatchInlineSnapshot(`
+      Object {
+        "contents": Object {
+          "kind": "plaintext",
+          "value": "int",
+        },
+        "range": Object {
+          "end": Object {
+            "character": 5,
+            "line": 2,
+          },
+          "start": Object {
+            "character": 4,
+            "line": 2,
+          },
+        },
+      }
+    `);
+
+    let result2 = await languageServer.sendRequest("textDocument/hover", {
+      textDocument: Types.TextDocumentIdentifier.create("file:///test.ml"),
+      position: Types.Position.create(2, 4),
+      verbosity: 2,
+    });
+
+    expect(result2).toMatchInlineSnapshot(`
+      Object {
+        "contents": Object {
+          "kind": "plaintext",
+          "value": "int",
+        },
+        "range": Object {
+          "end": Object {
+            "character": 5,
+            "line": 2,
+          },
+          "start": Object {
+            "character": 4,
+            "line": 2,
+          },
+        },
+      }
+    `);
+  });
```

Comment:
the rest of the tests are duplicated from textDocument/hover to check that my new command works properly. They can probably be removed to keep only ones playing with the verbosity once we know that things are working decently